src/firmware_new/boards/frankie_v1/io/proto_multicast_communication.h

```diff
@@ -0,0 +1,142 @@
+ *         |                               |         |                              |
+ *     +----> wait for PROTO_UPDATED event |     +----> wait for new proto on netif |
+ *     |   |                               |     |   |                              |
+ *     |   |  acquire lock                 |     |   |  acquire lock                |
+ *     |   |                               |     |   |                              |
+ *     |   |  serialize proto to buffer    |     |   |  deserialize buffer to proto |
+ *     |   |                               |     |   |                              |
+ *     |   |  release lock                 |     |   |  release lock                |
+ *     |   |                               |     |   |                              |
+ *     +----+ send buffer to group         |     +----+ notify RECEIVED_PROTO event |
+ *         |                               |         |                              |
+ *         +-------------------------------+         +------------------------------+
+ *
+ * @param arg The pointer to the allocated communication profile to use for the Task
+ * NOTE: These functions must run as a new Task
+ */
+void io_proto_multicast_sender_Task(void* arg);
+void io_proto_multicast_listener_Task(void* arg);
+
+/**
+ * LWIP automatically places MX_LWIP_Init() in a default task. That function
+ * call internally creates another tcpip task, and the default task just loops
+ * forever.
+ *
+ * Cube forces us to generate this function, in main.c where we mark it as __weak and
+ * define the implementation in this library (proto_mutlicast_communication). When the
+ * proto communication library is linked w/ the main frankie_v1 binary, the __weak
+ * reference is resolved to the "stronger" reference to the function defined below.
+ *
+ * We take over control so we can signal other networking tasks when MX_LWIP_Init()
+ * is finished running and the network link is configured and up.
+ */
+void io_proto_multicast_startNetworkingTask(void* arg);
+
+/**
+ * Initiallizes the proto_multicast_communication library
+ */
+void io_proto_multicast_communication_init(void);
+
+/**
+ * Create an ProtoMulticastCommunicationProfile
+ *
+ * Contains all the networking and protobuf information required to join a multicast
+ * group and communicate protobuf
+ *
+ * @param multicast_address [in] The multicast channel the robot should join
+ * @param port [in] The port to bind to
+ * @param protobuf_struct [in/out] The serialized proto in the incoming network packet,
+ *        will get deserialized into this struct in the listener task. The sender task
+ *        will serialize the protobuf_struct and send it over the network.
+ * @param message_fields [in] The nanopb message fields used to seriallize/deserialize
+ * @param message_max_size [in] The maximum known size of the protobuf.
+ *
+ * @return ProtoMulticastCommunicationProfile_t the profile to provide to the task
+ *
```

Comment:
nit: extra blank comment line can be removed

Reply:
removed